src/istore.c

```diff
@@ -976,3 +976,73 @@ istore_svals(PG_FUNCTION_ARGS)
 
     SRF_RETURN_DONE(funcctx);
 }
+
+/*
+ * A boolean function that supports `select '1=>1'::istore = '2=>2'::istore`
+ */
+PG_FUNCTION_INFO_V1(istore_eq);
+Datum
+istore_eq(PG_FUNCTION_ARGS)
+{
+    IStore *is1, *is2;
+    IStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
```

Comment:
I agree with the intended behavior for `NULL` arguments. Here we would not get an error though since the comparison function is declared `STRICT`. 
My point is, if someone uses the function in a non-intended/"illegal" way, there is no point in silently ignoring this – an exception should be raised instead.